modules/cxf/src/test/java/org/mule/module/cxf/issues/CfxClientRaceConditionTestCase.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.module.cxf.issues;
+
+import static javax.jms.DeliveryMode.NON_PERSISTENT;
+import static javax.jms.Session.AUTO_ACKNOWLEDGE;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.size.SmallTest;
+
+import javax.jms.BytesMessage;
+import javax.jms.Connection;
+import javax.jms.Destination;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+
+import org.apache.activemq.ActiveMQConnectionFactory;
+import org.apache.activemq.command.ActiveMQBytesMessage;
+import org.junit.Before;
+import org.junit.Test;
+
+@SmallTest
+public class CfxClientRaceConditionTestCase extends FunctionalTestCase
+{
+    private static final String AMDS_CXF_JAX_WS_CLIENT_EXPECTED_MESSAGE =
+            "<soap:Envelope xmlns:soap=\"http://www.w3.org/2003/05/soap-envelope\"><soap:Body/></soap:Envelope>";
+
+    private static final String INBOUND_TEST_QUEUE = "inboundTest";
+    
+    private static final String OUTBOUND_TEST_QUEUE = "cxfTest";
+    
+    private static final String ACTIVEMQ_CONNECTION_ADDRESS = "vm://localhost";
+    
+    private int totalMessages;
+
+    @Before
+    public void setup()
+    {
+        totalMessages = 10000;
+    }
+
+    @Test
+    public void testRaceCondition() throws Exception
+    {
+        cxfJmsTest(INBOUND_TEST_QUEUE, OUTBOUND_TEST_QUEUE, AMDS_CXF_JAX_WS_CLIENT_EXPECTED_MESSAGE);
```

Comment:
I've added a test using http as inbound endpoint. Load tests result in failure not so often as in the case of the case involving JMS inbound endpoint.